driver-mapping/src/main/java/com/datastax/driver/mapping/PropertyMapper.java

```diff
@@ -121,15 +132,27 @@ boolean isComputed() {
     }
 
     boolean isTransient() {
-        return hasAnnotation(Transient.class) ||
-                // If a property is both annotated and declared as transient in the class annotation, the property
-                // annotations take precedence (the property will not be transient)
-                classLevelTransients.contains(propertyName)
-                        && !hasAnnotation(PartitionKey.class)
-                        && !hasAnnotation(ClusteringColumn.class)
-                        && !hasAnnotation(Column.class)
-                        && !hasAnnotation(com.datastax.driver.mapping.annotations.Field.class)
-                        && !hasAnnotation(Computed.class);
+        // JAVA-1310: Make transient properties configurable at mapper level
+        // (should properties be transient by default or not)
+        switch (mapperConfiguration.getPropertyScanConfiguration().getPropertyMappingStrategy()) {
+            case BLACK_LIST:
+                return hasAnnotation(Transient.class) ||
```

Comment:
There is a pull request ( #750 ) that suggests to also consider fields annotated with the Java keyword `transient` – as they are probably not meant to be serialized nor persisted in any form. I think it's a valid point but the author has not accepted the CLA, so if you want to consider that here, I wouldn't be against it.